conans/client/generators/cmake_find_package.py

```diff
@@ -43,52 +48,256 @@ class CMakeFindPackageGenerator(Generator):
+        set({{ pkg_name }}_{{ comp_name }}_FRAMEWORK_DIRS {{ comp.framework_paths }})
+        set({{ pkg_name }}_{{ comp_name }}_FRAMEWORKS {{ comp.frameworks }})
+        set({{ pkg_name }}_{{ comp_name }}_BUILD_MODULES_PATHS {{ comp.build_modules_paths }})
+        set({{ pkg_name }}_{{ comp_name }}_DEPENDENCIES {{ comp.public_deps }})
+        set({{ pkg_name }}_{{ comp_name }}_LINKER_FLAGS_LIST
+                $<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,SHARED_LIBRARY>:{{ comp.sharedlinkflags_list }}>
+                $<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,MODULE_LIBRARY>:{{ comp.sharedlinkflags_list }}>
+                $<$<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>:{{ comp.exelinkflags_list }}>
+        )
+
+        {%- endfor %}
+
+
+        ########## FIND PACKAGE DEPENDENCY ##########################################################
+        #############################################################################################
+
+        include(CMakeFindDependencyMacro)
+
+        {%- for public_dep in pkg_public_deps %}
+
+        if(NOT {{ public_dep }}_FOUND)
+            find_dependency({{ public_dep }} REQUIRED)
+        else()
+            conan_message(STATUS "Conan: Dependency {{ public_dep }} already found")
+        endif()
+
+        {%- endfor %}
+
+
+        ########## FIND LIBRARIES & FRAMEWORKS / DYNAMIC VARS #######################################
+        #############################################################################################
+
+        {%- for comp_name, comp in components %}
+
+        ########## COMPONENT {{ comp_name }} FIND LIBRARIES & FRAMEWORKS / DYNAMIC VARS #############
+
+        set({{ pkg_name }}_{{ comp_name }}_FRAMEWORKS_FOUND "")
+        conan_find_apple_frameworks({{ pkg_name }}_{{ comp_name }}_FRAMEWORKS_FOUND "{{ '${'+pkg_name+'_'+comp_name+'_FRAMEWORKS}' }}" "{{ '${'+pkg_name+'_'+comp_name+'_FRAMEWORK_DIRS}' }}")
+
+        set({{ pkg_name }}_{{ comp_name }}_LIB_TARGETS "")
+        set({{ pkg_name }}_{{ comp_name }}_NOT_USED "")
+        set({{ pkg_name }}_{{ comp_name }}_LIBS_FRAMEWORKS_DEPS {{ '${'+pkg_name+'_'+comp_name+'_FRAMEWORKS_FOUND}' }} {{ '${'+pkg_name+'_'+comp_name+'_SYSTEM_LIBS}' }} {{ '${'+pkg_name+'_'+comp_name+'_DEPENDENCIES}' }})
+        conan_package_library_targets("{{ '${'+pkg_name+'_'+comp_name+'_LIBS}' }}"
+                                      "{{ '${'+pkg_name+'_'+comp_name+'_LIB_DIRS}' }}"
+                                      "{{ '${'+pkg_name+'_'+comp_name+'_LIBS_FRAMEWORKS_DEPS}' }}"
+                                      {{ pkg_name }}_{{ comp_name }}_NOT_USED
+                                      {{ pkg_name }}_{{ comp_name }}_LIB_TARGETS
+                                      ""
+                                      "{{ pkg_name }}_{{ comp_name }}")
+
+        set({{ pkg_name }}_{{ comp_name }}_LINK_LIBS {{ '${'+pkg_name+'_'+comp_name+'_LIB_TARGETS}' }} {{ '${'+pkg_name+'_'+comp_name+'_DEPENDENCIES}' }})
```

Comment:
The variable `{{pkg_name}}_{{compo_name}}_LIB_TARGETS` has already been linked with everything listed in `{{ pkg_name }}_{{ comp_name }}_LIBS_FRAMEWORKS_DEPS` (and also with system libraries that are wrongly listed in `libs`). So, `{{ pkg_name }}_{{ comp_name }}_LIB_TARGETS` should be all be need in line L177. Am I right?